newton/_src/sim/builder.py

```
@@ -1384,58 +1492,53 @@ def transform_mul(a, b):
+                # Handle array-like comparisons
+                if hasattr(defaults_match, "__iter__") and not isinstance(defaults_match, (str, bytes)):
+                    defaults_match = all(defaults_match)
+            except (ValueError, TypeError):
+                # If comparison fails, assume they're different
+                defaults_match = False
+
+            if not defaults_match:
+                raise ValueError(
+                    f"Custom attribute '{full_key}' default mismatch when merging builders: "
+                    f"existing={merged.default}, incoming={attr.default}"
                 )
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)